### DISCUSSION TITLE
Should an endpoint defined with an async function be requested using an async request library ?

### DISCUSSION THREAD

**Discussion by joaopfg** [fastapi/fastapi#6331] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
No example code
```

### Description

Suppose I define an endpoint with an async function:

```
...

@app.get("/endpoint")
async def f(...):
   ...

...

```

I noticed I can request it using the standard python` requests` library and it doesn't seem to cause problems.
Can it cause problem in some situation ? Should I use an async request library ? Am I missing some concept about async programming ?

### Operating System

Linux

### FastAPI Version

0.73.0

### Python Version

3.9.5

**Comment by samv-yazzoom** [comment]
How u do the request does not depend on the server, only on how u want the application that is talking to the server to work.
If u have a simple script that needs the response of the server there is not much point in using an async requests library.

On the other hand, there can be advantages if u want to do multiple I/O related things at the same time.
If you have a script that needs to do a request and read a file, then u could benefit from using asyncio libraries so that u can do the request and start loading the file whilst you are waiting on the network response.

Either way this issue is not related to the implementation of a fastapi application.

**Comment by joaopfg** [comment]
@samv-yazzoom But do I gain some benefit (in terms of async programming) if the endpoints are defined as async functions and the requests are made through the standard python requests library ? I mean, can the fact that I am using a non-async request library kill the benefits of defining the endpoint functions as async ?

In my case I am working on a relatively big project and the API is requested through different modules with possible concurrency. On server side, everything is already done with async programming (db related stuff, endpoint functions, auxiliar functions). But on the client side, things are done synchronously due to legacy code. In particular, requests are done using the standard python requests library. That's the reason for my question :)

**Comment by samv-yazzoom** [comment, marked as answer]
There is no difference for the server if the request was done using a sync or async library. The used library to call an http endpoint only makes for a difference in the application that uses the library (i.e. the client).

It's possible that u could leverage asyncio to do some other stuff in the client side applications whilst waiting for the server response; but this is completely unrelated to the implementation of the server.

**Comment by joaopfg** [comment]
@samv-yazzoom Thanks, I believe I understand it now. Closing the question
